include/crab/domains/region_domain.hpp

```diff
@@ -101,7 +102,8 @@ class region_domain final
   using base_linear_constraint_system_t =
       typename base_abstract_domain_t::linear_constraint_system_t;  
   using base_varname_allocator_t = typename Params::varname_allocator_t;
-
+  friend class mru_region_domain<Params>;
```

Comment:
Don't touch `region_domain.hpp`. I prefer for now to be `mru_region_domain` completely independent from `region_domain` at the expense of duplication. One design choice is that `mru_region_domain` could have `region_domain` as a private field to represent the non-cache state. However, since `mru_region_domain` needs to perform all the renaming stuff due to ghost variables I prefer not to call it.